source/app/models/url.rb

```diff
@@ -0,0 +1,26 @@
+require 'uri'
+require 'open-uri'
+
+class Url < ActiveRecord::Base
+  before_save :make_short_url
+  validates :real_url, presence: true
+  validate :test_url
+
+  private
+    def make_short_url
+      self.short_url = SecureRandom.hex(4) if self.short_url.nil? || self.short_url.empty?
+    end
+
+    def test_url
+      # validate url is http or https
+      uri = URI(self.real_url)
+      errors.add(:real_url, 'is not http or https') unless uri.scheme == 'http' || uri.scheme == 'https'
```

Comment:
I prefer using the accepted answer to this question: http://stackoverflow.com/questions/1805761/check-if-url-is-valid-ruby (under "Edit 2").